specs/latest/2.0/index.html

```diff
@@ -1092,6 +1094,52 @@ <h2 class="no-toc">Table of contents</h2>
         it is the responsibility of the WebGL API to ensure that data are accessed consistently. This applies
         even if the buffer is currently bound to a transform feedback binding point.
       </dd>
+      <dt>
+        <p class="idl-code">Promise&lt;ArrayBuffer&gt; getBufferSubDataAsync(GLenum target, GLintptr offset, ArrayBuffer returnedData)
```

Comment:
I think it's important for the application to allocate its own destination ArrayBuffer. It's been found that garbage generation is prohibitive for some applications, and there's already work in progress to eliminate garbage production for sophisticated applications (see #1940 ).

Race conditions are solved by only performing the write to returnedData when resolving the Promise. When the Promise callback is called, the data is available, but not before.